raiden-ts/src/path/epics.ts

```diff
@@ -1,48 +1,154 @@
   tap,
-  delay,
+  timeout,
+  withLatestFrom,
 } from 'rxjs/operators';
 import { fromFetch } from 'rxjs/fetch';
-import { isActionOf, ActionType } from 'typesafe-actions';
-import { bigNumberify, BigNumber } from 'ethers/utils';
-import { Zero, Two } from 'ethers/constants';
+import { ActionType, isActionOf } from 'typesafe-actions';
+import { BigNumber, bigNumberify, hexDataLength, hexlify, toUtf8Bytes } from 'ethers/utils';
+import { HashZero, Two, Zero } from 'ethers/constants';
 import { Event } from 'ethers/contract';
-import { isNil } from 'lodash';
+import { get, isNil } from 'lodash';
 
 import { RaidenAction } from '../actions';
 import { RaidenState } from '../state';
 import { RaidenEpicDeps } from '../types';
 import { getPresences$ } from '../transport/utils';
 import { messageGlobalSend } from '../messages/actions';
-import { PFSCapacityUpdate, MessageType } from '../messages/types';
-import { signMessage } from '../messages/utils';
+import { MessageType, PFSCapacityUpdate } from '../messages/types';
+import { MessageTypeId, signMessage } from '../messages/utils';
 import { channelDeposited } from '../channels/actions';
 import { ChannelState } from '../channels/state';
 import { channelAmounts } from '../channels/utils';
-import { Address, UInt, Int, decode } from '../utils/types';
-import { losslessStringify, losslessParse } from '../utils/data';
+import { Address, decode, Int, Signature, UInt } from '../utils/types';
+import { encode, losslessParse, losslessStringify } from '../utils/data';
 import { getEventsStream } from '../utils/ethers';
-import { pathFind, pathFound, pathFindFailed, pfsListUpdated } from './actions';
+import {
+  pathFind,
+  pathFindFailed,
+  pathFound,
+  pfsListUpdated,
+  udcBalanceFetch,
+  udcBalanceFetchFailed,
+  udcBalanceUpdate,
+} from './actions';
 import { channelCanRoute, pfsInfo, pfsListInfo } from './utils';
-import { PathResults, Paths } from './types';
+import { IOU, LastIOUResults, PathResults, Paths, PFS } from './types';
+import { concat } from 'ethers/utils/bytes';
+import { Signer } from 'ethers';
+
+const makeIOU = (sender: Address, receiver: Address, chainId: UInt<32>, oneToNAddress: Address) =>
+  ({
+    sender: sender,
+    receiver: receiver,
+    chain_id: chainId,
+    amount: Zero as UInt<32>,
+    one_to_n_address: oneToNAddress,
+    expiration_block: Zero as UInt<32>,
+    signature: HashZero as Signature,
+  } as IOU);
+
+const updateIOU = (iou: IOU, price: UInt<32>) =>
+  ({
+    ...iou,
+    amount: iou.amount.add(price),
+  } as IOU);
+
+const signIOU$ = (iou: IOU, signer: Signer) =>
+  from(signer.signMessage(
+    concat([
+      encode(iou.one_to_n_address, 20),
+      encode(iou.chain_id, 32),
+      encode(MessageTypeId.IOU, 32),
+      encode(iou.sender, 20),
+      encode(iou.receiver, 20),
+      encode(iou.amount, 32),
+      encode(iou.expiration_block, 32),
+    ]),
+  ) as Promise<Signature>).pipe(map(signature => ({ ...iou, signature } as IOU)));
```

Comment:
Same: return type vs type assertion, prefer the former, or in this case, `Observable<Signed<IOU>>` or equivalent